src/include/sof/math/exp_fcn_hifi4.h

```diff
@@ -0,0 +1,36 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2022 Intel Corporation. All rights reserved.
+ *
+ * Author: Shriram Shastry <malladi.sastry@linux.intel.com>
+ *
+ */
+#ifndef __SOF_MATH_EXPONENTIAL_HIFI4_H__
+#define __SOF_MATH_EXPONENTIAL_HIFI4_H__
+
+#include <stdint.h>
+#include <stddef.h>
+
+#define EXPONENTIAL_GENERIC
+#if defined(__XCC__)
+
+#include <xtensa/config/core-isa.h>
+#if XCHAL_HAVE_HIFI4 == 1
+#undef EXPONENTIAL_GENERIC
+#define EXPONENTIAL_HIFI4
+#endif
+
+#endif
+
+/* define constants */
+#define BIT_MASK_LOW_Q27P5 0x0000000008000000
+#define BIT_MASK_Q62P2 0x4000000000000000
+#define CONVERG_ERROR 28823037624320LL	// error smaller than 1e-4,1/2 ^ -44.7122876209085
+#define QUOTIENT_SCALE 0x400000000
+#define TERMS_Q23P9 0x800000
+#define LSHIFT_BITS 0x2000
```

Comment:
I'm not sure if macro BIT() works for 64 bits but at least for 32 bits value if the macro definiton is bitfield nature you could use it to make the code more clear. E:g. Here BIT(13).